tests/Unit/models/vpn/ipsec/ipsec.model.spec.ts

```diff
@@ -0,0 +1,1914 @@
+      await IPSec.createIPSecServerInterface(request, fwcloudProduct.ipsecServer.id);
+
+      const result = await db
+        .getSource()
+        .getRepository(IPSec)
+        .findOne({
+          where: { id: fwcloudProduct.ipsecServer.id },
+        });
+
+      expect(result).to.exist;
+    });
+  });
+
+  describe('updateIPSecServerInterface', () => {
+    it('should update interface and address for server', async () => {
+      const request: any = {
+        dbCon: db.getQuery(),
+        body: {
+          ipsec: fwcloudProduct.ipsecServer.id,
+          fwcloud: fwcloudProduct.fwcloud.id,
+          firewall: fwcloudProduct.firewall.id,
+          ipobj_g: fwcloudProduct.ipobjGroup.id,
+          install_name: StringHelper.randomize(10) + '.conf',
+        },
+      };
+
+      await IPSec.updateIPSecServerInterface(request);
+
+      const result = await db
+        .getSource()
+        .getRepository(IPSec)
+        .findOne({
+          where: { id: fwcloudProduct.ipsecServer.id },
+        });
+
+      expect(result).to.exist;
+    });
+  });
+
+  describe('moveToOtherFirewall', () => {
+    it('should move IPSec configurations to another firewall', async () => {
+      const newFirewallId = 9999; // Example new firewall ID
+      // Ensure the new firewall exists in the test setup
+      const newFirewall = await db
+        .getSource()
+        .getRepository(Firewall)
+        .findOne({ where: { id: newFirewallId } });
+
+      // Create the new firewall if it does not exist
+      if (!newFirewall) {
+        const newFirewallData = {
+          id: newFirewallId,
+          name: 'New Firewall',
+        };
+        await db.getSource().getRepository(Firewall).save(newFirewallData);
+      }
+
+      await IPSec.moveToOtherFirewall(db.getQuery(), fwcloudProduct.firewall.id, newFirewallId);
+
+      const result = await db
+        .getSource()
+        .getRepository(IPSec)
+        .findOne({
+          where: { id: fwcloudProduct.ipsecServer.id },
+        });
+
+      expect(result).to.exist;
+      expect(result).to.be.an('object');
+      expect(result).to.have.property('firewallId', newFirewallId);
+    });
+
+    it('should handle errors when moving to a non-existent firewall', async () => {
+      try {
+        await IPSec.moveToOtherFirewall(db.getQuery(), fwcloudProduct.firewall.id, -9999);
+      } catch (error) {
+        expect(error).to.exist;
+      }
+    });
+  });
+
+  describe('getConfigFilename', () => {
+    it('should create a config file of a IPSec server', async () => {
+      const result = await IPSec.getConfigFilename(db.getQuery(), fwcloudProduct.firewall.id);
+
+      expect(result).to.exist;
+      expect(result).to.be.a('string');
+    });
+
+    it('should return default config filename when invalid firewall', async () => {
+      const result = await IPSec.getConfigFilename(db.getQuery(), -9999);
+
+      expect(result).to.exist;
+      expect(result).to.be.a('string');
+    });
+  });
```

Comment:
The test only checks that a string is returned, but not that it's the default name.